# INFO7390 - Advance Data Science and Architecture

## Project Title: Job Recommendation System
### Teammates:
1. Aniruddha Tambe
2. Shubhankar Salvi
3. Sangram Vuppula

## Importing packages:

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import missingno as msno
from icecream import ic
import os
import re
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.figure_factory as ff
from kaleido.scopes.plotly import PlotlyScope
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import OneHotEncoder
from sklearn.feature_selection import SelectKBest
import warnings
warnings.filterwarnings(action='ignore')
pd.options.display.max_columns = 100

# plotly settings and functions
scope = PlotlyScope(plotlyjs="https://cdn.plot.ly/plotly-latest.min.js")
pio.templates.default = 'plotly_white'

## Dataset:

1. Stack Overflow 2018 Developer Survey - Individual responses on the 2018 Developer Survey fielded by Stack Overflow

https://www.kaggle.com/stackoverflow/stack-overflow-2018-developer-survey#survey_results_public.csv

2. U.S. Technology Jobs on Dice.com - 22,000 US-based Technology Job Listings

https://www.kaggle.com/PromptCloudHQ/us-technology-jobs-on-dicecom

In [2]:
survey = pd.read_csv("./dataset/survey_results_public.csv")

## Preliminary data summary

In [3]:
# List all column names
all_col_names = survey.columns.values.tolist()
print('Number of columns: ',len(all_col_names))
#print(all_col_names)

Number of columns:  129


In [4]:
# Get categorical columns
cat_cols = [col for col in survey.columns if survey[col].dtype.name=="object"]
float_cols = [col for col in survey.columns if survey[col].dtype.name=="float64"]
int_cols = [col for col in survey.columns if survey[col].dtype.name=="int64"]
print('Number of categorical columns: ',len(cat_cols))
print('Number of float columns: ',len(float_cols))
print('Number of int columns: ',len(int_cols))

Number of categorical columns:  87
Number of float columns:  41
Number of int columns:  1


## Findings

1. Number of categorical columns:  87
2. Number of float columns:  41
3. Number of int columns:  1
4. Datatypes found: float64(41), int64(1), object(87)
5. Rows x Columns: 98855 x 129

## Dropping irrelevant columns

In [5]:
attrToDrop=[]
survey=survey[survey.columns.difference(attrToDrop)]
print("Number of columns dropped: ",len(attrToDrop))

Number of columns dropped:  102


In [6]:
# Get categorical columns
cat_cols = [col for col in survey.columns if survey[col].dtype.name=="object"]
float_cols = [col for col in survey.columns if survey[col].dtype.name=="float64"]
int_cols = [col for col in survey.columns if survey[col].dtype.name=="int64"]
print('Number of categorical columns: ',len(cat_cols))
print('Number of float columns: ',len(float_cols))
print('Number of int columns: ',len(int_cols))
print('Total number of columns: ',len(survey.columns))

Number of categorical columns:  43
Number of float columns:  29
Number of int columns:  1
Total number of columns:  73


## Export survey to csv

In [13]:
survey.to_csv('./dataset/survey_dropped_columns.csv',index=False)

## Survey